metrics.go

```diff
@@ -0,0 +1,52 @@
+package ipam
+
+import (
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	prometheusNamespace = "ipam"
+)
+
+var (
+	subnetCounter = prometheus.NewGauge(
+		prometheus.GaugeOpts{
```

Comment:
It's not _super_ straightforward, so I don't want to do it here, but monitoring network fragmentation / number of networks left feels right - see https://github.com/giantswarm/giantswarm/issues/1756 for tracking